packages/components/src/components/Ansi/Ansi.js

```diff
@@ -0,0 +1,268 @@
+  };
+
+  const setFGColor = (type, command) => {
+    properties.foregroundColor = getTermColor(type, command);
+  };
+
+  const setBGColor = (type, command) => {
+    properties.backgroundColor = getTermColor(type, command);
+  };
+
+  const setFGColor256 = commandStack => {
+    properties.foregroundColor = getXtermColor(commandStack);
+  };
+
+  const setBGColor256 = commandStack => {
+    properties.backgroundColor = getXtermColor(commandStack);
+  };
+
+  const setProperties = {
+    0: () => reset(),
+    1: () => enable('bold'),
+    3: () => enable('italic'),
+    4: () => enable('underline'),
+    8: () => enable('conceal'),
+    9: () => enable('cross'),
+
+    21: () => disable('bold'),
+    22: () => disable('bold'),
+    23: () => disable('italic'),
+    24: () => disable('underline'),
+    28: () => disable('conceal'),
+    29: () => disable('cross'),
+
+    30: () => setFGColor(0),
+    31: () => setFGColor(1),
+    32: () => setFGColor(2),
+    33: () => setFGColor(3),
+    34: () => setFGColor(4),
+    35: () => setFGColor(5),
+    36: () => setFGColor(6),
+    37: () => setFGColor(7),
+    38: s => setFGColor256(s),
+    39: () => setFGColor(9),
+
+    40: () => setBGColor(0),
+    41: () => setBGColor(1),
+    42: () => setBGColor(2),
+    43: () => setBGColor(3),
+    44: () => setBGColor(4),
+    45: () => setBGColor(5),
+    46: () => setBGColor(6),
+    47: () => setBGColor(7),
+    48: s => setBGColor256(s),
+    49: () => setBGColor(9),
+
+    90: () => setFGColor(0, 'l'),
+    91: () => setFGColor(1, 'l'),
+    92: () => setFGColor(2, 'l'),
+    93: () => setFGColor(3, 'l'),
+    94: () => setFGColor(4, 'l'),
+    95: () => setFGColor(5, 'l'),
+    96: () => setFGColor(6, 'l'),
+    97: () => setFGColor(7, 'l'),
+    99: () => setFGColor(9, 'l'),
+
+    100: () => setBGColor(0, 'l'),
+    101: () => setBGColor(1, 'l'),
+    102: () => setBGColor(2, 'l'),
+    103: () => setBGColor(3, 'l'),
+    104: () => setBGColor(4, 'l'),
+    105: () => setBGColor(5, 'l'),
+    106: () => setBGColor(6, 'l'),
+    107: () => setBGColor(7, 'l'),
+    109: () => setBGColor(9, 'l')
+  };
+
+  const setStyle = (command, stack) => {
+    if (setProperties[command]) {
+      setProperties[command](stack);
+    }
+  };
+
+  const evaluateCommandStack = stack => {
+    const command = stack.shift();
+    if (!command) {
+      return;
+    }
+    setStyle(command, stack);
+    evaluateCommandStack(stack);
+  };
+
+  const linkify = (str, styleObj) => {
+    if (!str) {
+      return null;
+    }
+    const matches = linkifyIt.match(str);
+    if (!matches) {
+      return <span style={styleObj}>{str}</span>;
+    }
+    const elements = [];
+    let offset = 0;
+    matches.forEach(match => {
+      if (match.index > offset) {
+        const string = str.substring(offset, match.index);
+        elements.push(<span style={styleObj}>{string}</span>);
+      }
+      elements.push(
+        <a href={match.url} style={styleObj}>
```

Comment:
open in new window and don't give child window access to the opener object

```suggestion
        <a href={match.url} style={styleObj} target="_blank" rel="noopener noreferrer">
```